modules/js-and-css/speculation-rules/class-plsr-url-pattern-prefixer.php

```diff
@@ -69,12 +74,23 @@ public function prefix_path_pattern( string $path_pattern, string $context = 'ho
 			return $path_pattern;
 		}
 
-		// If the path already starts with the context path (including '/'), there is nothing to prefix.
-		if ( str_starts_with( $path_pattern, $this->contexts[ $context ] ) ) {
-			return $path_pattern;
+		// In the event that the context path contains a :, ? or # (which can cause the URL pattern parser to
+		// switch to another state, though only the latter two should be percent encoded anyway), we need to
+		// additionally enclose it in grouping braces. The final forward slash (trailingslashit ensures there is
+		// one) affects the meaning of the * wildcard, so is left outside the braces.
+		$context_path         = $this->contexts[ $context ];
+		$escaped_context_path = $context_path;
+		if ( strcspn( $context_path, ':?#' ) !== strlen( $context_path ) ) {
```

Comment:
Clever. I would have done `str_contains( $context_path, ':' ) || str_contains( $context_path, '?' ) || str_contains( $context_path, '#' )` but clearly what you've done is faster.

Reply:
Or rather:

```suggestion
		if ( preg_match( '/[:?#]/', $context_path ) ) {
```

This would be the more common way to do it (and perhaps more readable?) There are 31 uses of `strcspn()` in core, whereas there are 635 of `preg_replace()`.

I don't feel strongly either way.

Reply:
It's probably being a C++ developer by day, but my mild preference is for the most efficient option if the options are similar in readability (though obviously neither is going to have a massive effect on your WordPress performance).

Happy to change if you felt more strongly; it seems not, so leaving this as-is for now.